File: UnitB/ch04/zhypotheistest-template.jl
"""
question :
page :lock5 stat page 
dataset:
describe:
将一组相当于人类 70 岁的小鼠随机分为两组, 分别输入年轻小鼠的血浆和年老小鼠的血浆,
之后用在 90min 的窗口内测量小鼠的跑步时间

实验流程

1. 输入观测数据
2.t检验
3. 分析结果
   
4. 结论:

"""


include("../../utils.jl")


  df= (let str="YoungBlood"
      df=load_data(str)
  end)



ages=levels(df[!,:Plasma])
gdf=groupby(df,:Plasma)  #g1 输入老龄鼠血浆, g2组输入年轻鼠血浆
EqualVarianceTTest(gdf[1][!,:Runtime ],gdf[2][!,:Runtime])

File: marimo/ch04/zhypotheistest-template.py
import marimo

__generated_with = "0.24.0"
app = marimo.App(width="medium")


@app.cell
def _():
    import marimo as mo
    import sys
    from pathlib import Path
    import numpy as np
    import polars as pl
    import matplotlib.pyplot as plt
    from scipy import stats
    from great_tables import GT
    _p = Path(mo.notebook_location()).resolve().parent
    while _p != _p.parent and not (_p / "lock5.py").exists():
        _p = _p.parent
    if str(_p) not in sys.path:
        sys.path.insert(0, str(_p))
    import lock5
    return mo, sys, Path, np, pl, plt, stats, GT, lock5


@app.cell
def _(mo):
    mo.md(r"""# Hypothesis test template — YoungBlood plasma

Port of `UnitB/ch04/zhypotheistest-template.jl`. Two-sample equal-variance
t-test of `Runtime` between the `Old` and `Young` plasma groups.""")
    return


@app.cell
def _(lock5, np, pl):
    df = lock5.load_data("YoungBlood")
    gdf = df.group_by("Plasma", maintain_order=True).agg(pl.col("Runtime"))
    g1 = np.asarray(gdf.row(0)[1], dtype=float)
    g2 = np.asarray(gdf.row(1)[1], dtype=float)
    return df, g1, g2, gdf


@app.cell
def _(g1, g2, lock5, mo):
    res = lock5.equal_var_ttest(g1, g2)
    mo.Html(lock5.gt_ttest(res).as_raw_html())
    return res


if __name__ == "__main__":
    app.run()
